StackExchange/Stacks
# chore(links): update visited links colors w/ a11y considerations

This PR addresses [STACKS-481](https://stackoverflow.atlassian.net/browse/STACKS-481) (update visited link color) and [STACKS-646](https://stackoverflow.atlassian.net/browse/STACKS-646) (underline links within in paragraphs to meet [WCAG 2.2 SC 1.4.1](https://www.w3.org/WAI/WCAG22/Techniques/failures/F73)). It should also incidentally resolve [STACKS-480](https://stackoverflow.atlassian.net/browse/STACKS-480) (improve contrast between unvisited and visited links on Meta).

### Changes in this PR

- Updates the visited link style to `purple-500` (and `purple-600` when visited and hovering)
- Adds an underline to elements with `.s-link` when within a paragraph
    - It also updates the docs to mimic this style

### To test visited link color

1. Go to the "[Single link examples](https://deploy-preview-1740--stacks.netlify.app/product/components/links/#single-link-examples)" section of the links docs
2. Look at the "visited" link example
3. **Bonus**: Try it out in different color modes
4. **Double bonus**: Open a link in the docs and observe it change from blue to purple

### To test underline

We don't include an `.s-link` within a paragraph anywhere in our docs, but we did apply the same styling to links in the documentation copy. You can see those by doing the following:

1. Visit a docs page that includes links within copy (e.g. the [accessibility page](https://deploy-preview-1740--stacks.netlify.app/product/foundation/accessibility/))
2. Check out those underlined links 😎 

### Screenshots

<details>
<summary>Light mode</summary>

#### Default
![Screen Shot 2024-07-30 at 17 38 14](https://github.com/user-attachments/assets/e2ac5e3c-ded9-4a9b-9036-3c3334563899)

#### High contrast
![20240730-174218](https://github.com/user-attachments/assets/dc5c3889-62b5-41f2-bfb8-a796cfb48313)

</details>

<details>
<summary>Dark mode</summary>

#### Default
![20240730-174009](https://github.com/user-attachments/assets/dbd5b5f2-1c3e-4ba0-9a1a-e7000c2b9993)

#### High contrast
![Screen Shot 2024-07-30 at 17 41 25](https://github.com/user-attachments/assets/63203038-aa62-40fa-9280-d4aa3d8643bb)

</details>

---

### History

[STACK-481](https://stackoverflow.atlassian.net/browse/STACKS-481)

In https://github.com/StackExchange/Stacks/pull/1735, we updated visited link styles. @giamir [later pointed out](https://github.com/StackExchange/Stacks/pull/1738#issuecomment-2124075186) that we should consider [WCAG SC 1.4.1](https://www.w3.org/WAI/WCAG22/Understanding/use-of-color.html), which suggests that links within otherwise unlinked text needs to be differentiated from surrounding text either by some means other than color (underlining, bolding, etc) or differentiate significantly by both hue and lightness (see [comment](https://github.com/StackExchange/Stacks/pull/1738#issuecomment-2125690020)). Also see [A11Y-108](https://stackoverflow.atlassian.net/browse/A11Y-108)

Because link styles may require more work/consideration than was performed in https://github.com/StackExchange/Stacks/pull/1735, we decided to [revert that PRs merge](https://github.com/StackExchange/Stacks/commit/43b6c22248646628d7cc4663c639509f0985bfec) to avoid shipping a half-baked change. In this PR, we'll resolve any lingering accessibility issues with link color and then ship it out.

cc @giamir @CGuindon
